code/hud/hudconfig.cpp

```diff
@@ -851,6 +859,44 @@ void hud_config_popup_flag_clear(int i)
 	}
 }
 
+void hud_config_set_mouse_coords(int gauge_config, int x1, int x2, int y1, int y2) {
+	HC_gauge_mouse_coords[gauge_config] = {x1, x2, y1, y2};
+}
+
+void hud_config_convert_coords(int x, int y, float scale, int& outX, int& outY)
+{
+	outX = HC_gauge_coordinates[0] + static_cast<int>(x * scale);
+	outY = HC_gauge_coordinates[2] + static_cast<int>(y * scale);
+}
+
+void hud_config_convert_coords(float x, float y, float scale, float& outX, float& outY)
+{
+	outX = HC_gauge_coordinates[0] + x * scale;
+	outY = HC_gauge_coordinates[2] + y * scale;
+}
+
+void hud_config_get_scale(int baseW, int baseH, float& outScale)
+{
+	// Determine the scaling factor
+	float scaleX = static_cast<float>(HC_gauge_coordinates[4]) / baseW;
+	float scaleY = static_cast<float>(HC_gauge_coordinates[5]) / baseH;
+
+	// Use the smallest scale factor
+	outScale = std::min(scaleX, scaleY);
+}
+
+void hud_config_convert_coord_sys(int x, int y, int baseW, int baseH, int& outX, int& outY, float& outScale)
+{
```

Comment:
I won't request you to change this now, given that it's only really a subjective style difference.
But given that we've now got C++17, consider returning tuples of values that you can then structured bind instead of having a bajillion variable references to move out to again:
```cpp
std::tuple<int, int, float> void hud_config_convert_coord_sys(int x, int y, int baseW, int baseH) {
   return std::make_tuple(/* the values */);
}

...

const auto& [outX, outY, outScale] = hud_config_convert_coord_sys(...);
```
That way, it's less painful trying to figure out what goes out, what goes in, which outparams might need to be in a certain state. You just get back a bunch of params that you can then, in one line, assign to a bunch of variables.

Reply:
No, I'm all for clarity and cleanup. Especially during the framework phase. Done.